TogglTrack/Networking/URLSessionExtension.swift

```diff
@@ -18,19 +18,60 @@ extension URLSession: URLSessionProtocol
 {
     public func load<A>(_ endpoint: Endpoint<A>) -> AnyPublisher<A, Error>
     {
-        return dataTaskPublisher(for: endpoint.request)
+        let publisher = dataTaskPublisher(for: endpoint.request).share()
+
+        return publisher
+            .catch {_ in
+                publisher.delay(for: DispatchQueue.SchedulerTimeType.Stride(integerLiteral: 4),
+                                scheduler: DispatchQueue.global(qos: .default))
+            }
             .retry(3)
```

Comment:
Not sure why this fails, but the way you are doing retries is a little funky. This one seems to work:

```suggestion
        return dataTaskPublisher(for: endpoint.request).share()
            .retryWithDelay()
```

It needs this extension somewhere:
```swift
extension Publisher {
    func retryWithDelay<T, E>(retries: Int = 3, delay: Int = 4)
        -> Publishers.Catch<Self, AnyPublisher<T, E>> where T == Self.Output, E == Self.Failure
    {
        return self.catch { error -> AnyPublisher<T, E> in
            return Publishers.Delay(
                upstream: self,
                interval: .seconds(delay),
                tolerance: 1,
                scheduler: DispatchQueue.global()
            )
            .retry(retries)
            .eraseToAnyPublisher()
        }
    }
}
```